### ISSUE TITLE
FileConfig failIfMissing parameter is used wrongly

### ISSUE THREAD

**Issue by twinprime** [kohesive/klutter#38]
I think the intended behavior of the class is to throw an exception if the given file is not found by default (which is inline with the spirit of Typesafe config) but it is currently doing the reverse.

```
open class FileConfig(val file: File, val failIfMissing: Boolean = true) : ConfigLoader() {
    override fun load(): Config {
        val options = ConfigParseOptions.defaults().setAllowMissing(failIfMissing)
        return ConfigFactory.parseFileAnySyntax(file, options)
    }
}
```

I think it should be corrected as such but it will cause backward compatibility issues:

val options = ConfigParseOptions.defaults().setAllowMissing(**!failIfMissing**)

The other option is to rename the parameter to "allowMissing" in order to maintain backward compatibility.

Same issue for many other classes in the same file.

**Comment by apatrida** [comment]
thanks, I think it is best to have it work by how the parameter is named, since most people probably coded to the name and not the actual functionality (if they tested it at all, they likely would have reported it as you did).  I'll go into a new version with a feature bump and anyone watching this issue will see it go into the release.

**Comment by apatrida** [comment]
Fixed in 2.4.0 as mentioned above, so the meaning of the flags now matches the naming, you need to update your code if you changed it for actual behaviour versus how the API was defined.  

2.4.0 is being released now.
